pkg/daemons/executor/executor.go

```diff
@@ -103,7 +103,7 @@ func (e ETCDConfig) ToConfigFile(extraArgs []string) (string, error) {
 				} else if time, err := time.ParseDuration(extraArg[1]); err == nil && (strings.Contains(lowerKey, "time") || strings.Contains(lowerKey, "duration") || strings.Contains(lowerKey, "interval") || strings.Contains(lowerKey, "retention")) {
 					// auto-compaction-retention is either a time.Duration or int, depending on version. If it is an int, it will be caught above.
 					s[key] = time
-				} else if err := yaml.Unmarshal([]byte(extraArg[1]), &stringArr); err == nil {
+				} else if err := yaml.Unmarshal([]byte(extraArg[1]), &stringArr); err == nil && (len(extraArg[1]) > 0) {
```

Comment:
I think this belongs up on line 97? `if len(extraArg) == 2 && len(extraArg[1]) > 0`

Reply:
If I do that, the parameter will not be inserted into the `s` map (https://github.com/jordanorc/k3s/blob/a0a4885eafd39163e417601029bd19073ced7cdd/pkg/daemons/executor/executor.go#L107) and will never be overridden with an empty string.

Reply:
OK but what is your intent here - if the arg value is an empty string, do you want the config value set to an empty string, or do you want to ensure that the key is removed so that it takes the default value?

What you're doing here will set the value to an empty string, which is valid for config entries that expect a string value. That is not all of them, as shown by the special handling for integer and time/duration fields.

If your intent is to get the **default** value by removing the key from the config, then a different approach is required.